experiment_buddy/experiment_buddy.py

```diff
@@ -50,11 +52,16 @@ def register(config_params):
     parser = argparse.ArgumentParser()
     parser.add_argument('_ignored', nargs='*')
 
+    if any(k.startswith(wandb_escape) for k in config_params.keys()):
+        raise NameError(f"{wandb_escape} is a reserved prefix")
+
     for k, v in config_params.items():
-        if k.startswith(wandb_escape):
-            raise NameError(f"{wandb_escape} is a reserved prefix")
         if _is_valid_hyperparam(k, v):
-            parser.add_argument(f"--{k}", f"--^{k}", type=type(v), default=v)
+            new_arg = partial(parser.add_argument, f"--{k}", f"--^{k}", default=v)
+            if isinstance(v, bool):
+                new_arg(type=lambda x: bool(strtobool(x)), choices=[True, False])
```

Comment:
why do you have to cast to boolean if the function is called `strtobool`?

furthermore it seems a similar feature is coming.  https://docs.python.org/3/library/argparse.html#action

Reply:
This is the only solution I've found that:
 - return False when the input is explicit 'False'
 - return True when the input is explicit 'True'
 - fails with the input is empty, giving an explanation of the correct input from argparse
 - fails with the input is neither 'True' or 'False', giving an explanation of the correct input from argparse
 - I've mixed feelings with using `--foo/--no-foo`
 - About the module `BooleanOptionalAction` I'm optimistic, but not as "all the users are going to use python 3.9" optimistic

Also: 
 `strtobool` returns `0|1`, `bool` converts it to the correct type

Reply:
XD why did you name a function strtobool to return a binary and then cast it, I think it should return a boolean.

Reply:
ahhhhh it's builtin. Mybad,